From 54775ceec6d2574fe3bb6e29dd3f98065d0162c0 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Fri, 11 Sep 2015 17:03:00 +0800
Subject: [PATCH 0438/5242] MLK-11530-04 dmaengine: imx-sdma: add hdmi audio
 support

commit  26f37f295318f7bcee63ee2d69b037b78e21b409 from
https://source.codeaurora.org/external/imx/linux-imx.git

No need to alloc bd for hdmi audio, and the period len of hdmi audio is
0, so add constraint for it.

And correct per_address and per_address2 for hdmi audio.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index f0bacce..1831a78 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -1655,8 +1655,8 @@ static int sdma_config(struct dma_chan *chan,
 			SDMA_WATERMARK_LEVEL_HWML;
 		sdmac->word_size = dmaengine_cfg->dst_addr_width;
 	} else if (sdmac->peripheral_type == IMX_DMATYPE_HDMI) {
-			sdmac->per_address = dmaengine_cfg->src_addr;
-			sdmac->per_address2 = dmaengine_cfg->dst_addr;
+			sdmac->per_address = dmaengine_cfg->dst_addr;
+			sdmac->per_address2 = dmaengine_cfg->src_addr;
 			sdmac->watermark_level = 0;
 	} else if (dmaengine_cfg->direction == DMA_MEM_TO_MEM) {
 			sdmac->word_size = dmaengine_cfg->dst_addr_width;
-- 
1.7.9.5

